plugins/bulk_upload/drop_folder_xml/plugins/providers/ap_feed/ApFeedDropFolderPlugin.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * @package plugins.ApFeedDropFolder
+ */
+class ApFeedDropFolderPlugin extends KalturaPlugin implements IKalturaPlugin, IKalturaPending, IKalturaObjectLoader, IKalturaEnumerator, IKalturaApplicationTranslations
+{
+	const PLUGIN_NAME = 'ApFeedDropFolder';
+	
+	const DROP_FOLDER_PLUGIN_NAME = 'dropFolder';
+	/**
+	 * @return array
+	 */
+	public static function getTranslations($locale)
+	{
+		$array = array();
+		
+		$langFilePath = __DIR__ . "/config/lang/$locale.php";
+		if(!file_exists($langFilePath))
+		{
+			$default = 'en';
+			$langFilePath = __DIR__ . "/config/lang/$default.php";
+		}
+		
+		KalturaLog::info("Loading file [$langFilePath]");
+		$array = include($langFilePath);
+		
+		return array($locale => $array);
+	}
+	
+	/**
+	 * Returns a list of enumeration class names that implement the baseEnumName interface.
+	 * @param string $baseEnumName the base implemented enum interface, set to null to retrieve all plugin enums
+	 * @return array<string> A string listing the enum class names that extend baseEnumName
+	 */
+	public static function getEnums($baseEnumName = null)
+	{
+		if (!$baseEnumName)
+		{
+			return array('ApFeedDropFolderType');
+		}
+		if ($baseEnumName == 'DropFolderType')
+		{
+			return array('ApFeedDropFolderType');
+		}
+		
+		return array();
+	}
+	
+	/**
+	 * Returns an object that is known only to the plugin, and extends the baseClass.
+	 *
+	 * @param string $baseClass The base class of the loaded object
+	 * @param string $enumValue The enumeration value of the loaded object
+	 * @param array $constructorArgs The constructor arguments of the loaded object
+	 * @return object
+	 */
+	public static function loadObject($baseClass, $enumValue, array $constructorArgs = null)
+	{
+		switch ($baseClass)
+		{
+			case 'KDropFolderEngine':
+				if ($enumValue == KalturaDropFolderType::AP_FEED)
+				{
+					return new KApFeedDropFolderEngine();
+				}
+				break;
+			case ('KalturaDropFolderFile'):
+				if ($enumValue == self::getDropFolderTypeCoreValue(ApFeedDropFolderType::AP_FEED) )
+				{
+					return new KalturaFeedDropFolderFile();
+				}
+				break;
+			case ('KalturaDropFolder'):
+				if ($enumValue == self::getDropFolderTypeCoreValue(ApFeedDropFolderType::AP_FEED) )
+				{
+					return new KalturaApFeedDropFolder();
+				}
+				break;
+			case 'kDropFolderXmlFileHandler':
+				if ($enumValue == self::getDropFolderTypeCoreValue(ApFeedDropFolderType::AP_FEED))
+				{
+					return new kDropFolderFeedXmlFileHandler();
+				}
+				break;
+			case 'Form_DropFolderConfigureExtend_SubForm':
+				if ($enumValue == Kaltura_Client_DropFolder_Enum_DropFolderType::AP_FEED)
+				{
+					return new Form_ApFeedDropFolderConfigureExtend_SubForm();
+				}
+				break;
+			case 'Kaltura_Client_DropFolder_Type_DropFolder':
+				if ($enumValue == Kaltura_Client_DropFolder_Enum_DropFolderType::AP_FEED)
+				{
+					return new Kaltura_Client_ApFeedDropFolder_Type_ApFeedDropFolder();
+				}
+				break;
```

Comment:
What should be the default behaviour of this switch case? should it print an message? throw exception do-nothing?
Please add default to the switch.

Reply:
@MosheMaorKaltura there is no default on any of the plugin classes that implement the loadObject through a switch case. The reason is that this method is being called from the KalturaPluginManager, which iterates over all the plugins that implement the loadObject function. If all plugins had a default logging a message, you would see several KalturaLog messages all reading "Plugin xyz doesn't handle this class". If you threw an exception, you would prevent the PluginManager from processing the next plugin class which **might** be the correct one. Any logic added in a default case would be bad here so there is no default.